src/main/java/net/minecraftforge/common/extensions/IForgeBlockState.java

```diff
@@ -603,21 +608,47 @@ default boolean shouldDisplayFluidOverlay(BlockAndTintGetter level, BlockPos pos
     }
 
     /**
-     * Returns the state that this block should transform into when right clicked by a tool.
-     * For example: Used to determine if an axe can strip, a shovel can path, or a hoe can till.
-     * Return null if vanilla behavior should be disabled.
+     * Returns the tool modification result describing what should happen to this block when right-clicked by a tool.
+     * For example: Used to determine if {@link ToolActions#AXE_STRIP an axe can strip},
+     * {@link ToolActions#SHOVEL_FLATTEN a shovel can path}, or {@link ToolActions#HOE_TILL a hoe can till}.
+     * If {@link ToolModificationResult#passed()} is true, a null tool modified state will fall back on default behavior.
+     * A non-null tool modified state will update the block accordingly.
+     * If {@link ToolModificationResult#failed()} is true, the action will be cancelled regardless of the tool modified state.
+     *
+     * @param context The use on context that the action was performed in
+     * @param toolAction The action being performed by the tool
+     * @return The result holder containing the tool modified state attached to a pass or failure state
+     */
+    @NotNull
+    default ToolModificationResult getToolModificationResult(UseOnContext context, ToolAction toolAction)
+    {
+        ToolModificationResult toolModificationResult = ForgeEventFactory.onToolUse(context, self(), toolAction);
+        if (toolModificationResult.failed() || (toolModificationResult.passed() && toolModificationResult.toolModifiedState() != null))
+            return toolModificationResult;
+
+        return self().getBlock().getToolModificationResult(self(), context, toolAction);
+    }
+
+    /**
+     * Returns the state that this block should transform into when right-clicked by a tool.
+     * For example: Used to determine if {@link ToolActions#AXE_STRIP an axe can strip},
+     * {@link ToolActions#SHOVEL_FLATTEN a shovel can path}, or {@link ToolActions#HOE_TILL a hoe can till}.
+     * Returns {@code null} if the default behavior should be disabled.
      *
      * @param level The level
      * @param pos The block position in level
      * @param player The player clicking the block
      * @param stack The stack being used by the player
      * @param toolAction The tool type to be considered when performing the action
      * @return The resulting state after the action has been performed
+     * @deprecated Use {@link #getToolModificationResult(net.minecraft.world.item.context.UseOnContext, ToolAction)} instead
      */
+    @SuppressWarnings("removal")
     @Nullable
+    @Deprecated(forRemoval = true, since = "1.18.2")
```

Comment:
Was the `since` field not for the version it is supposed to be deleted in?

Reply:
We had discussion on this previously and we ended up deciding on switching to using `since` properly, where you put the version it was deprecated in (since XXX). And the next major version removes it.

Reply:
Okiedokie, I made a note :D